apps/antalmanac/src/components/RightPane/AddedCourses/AddedCoursePane.tsx

```diff
@@ -1,277 +1,328 @@
             category: analyticsEnum.addedClasses.title,
-            action: analyticsEnum.addedClasses.actions.OPEN,
+            action: analyticsEnum.addedClasses.actions.CLEAR_SCHEDULE,
         });
+    }
+}
+
+function createCopyHandler(index: number) {
+    return () => {
+        copySchedule(index);
     };
+}
 
-    componentWillUnmount() {
-        AppStore.removeListener('addedCoursesChange', this.loadCourses);
-        AppStore.removeListener('customEventsChange', this.loadCustomEvents);
-        AppStore.removeListener('currentScheduleIndexChange', this.loadCourses);
-        AppStore.removeListener('currentScheduleIndexChange', this.loadCustomEvents);
-        AppStore.removeListener('scheduleNamesChange', this.loadScheduleNames);
-        AppStore.removeListener('scheduleNotesChange', this.loadScheduleNote);
-    }
+function SkeletonSchedule() {
+    const [skeletonSchedule, setSkeletonSchedule] = useState(AppStore.getSkeletonSchedule());
+
+    useEffect(() => {
+        const updateSkeletonSchedule = () => {
+            setSkeletonSchedule(AppStore.getSkeletonSchedule());
+        };
+
+        AppStore.on('skeletonScheduleChange', updateSkeletonSchedule);
 
-    loadCourses = () => {
-        const currentCourses = AppStore.schedule.getCurrentCourses();
-        const formattedCourses: CourseWithTerm[] = [];
-
-        for (const course of currentCourses) {
-            let formattedCourse: CourseWithTerm | undefined = formattedCourses.find(
-                (needleCourse) =>
-                    needleCourse.courseNumber === course.courseNumber && needleCourse.deptCode === course.deptCode
-            );
-
-            if (formattedCourse) {
-                formattedCourse.sections.push({
-                    ...course.section,
-                });
-            } else {
-                formattedCourse = {
-                    term: course.term,
-                    deptCode: course.deptCode,
-                    courseComment: course.courseComment,
-                    prerequisiteLink: course.prerequisiteLink,
-                    courseNumber: course.courseNumber,
-                    courseTitle: course.courseTitle,
-                    sections: [
-                        {
-                            ...course.section,
-                        },
-                    ],
-                };
-                formattedCourses.push(formattedCourse);
+        return () => {
+            AppStore.off('skeletonScheduleChange', updateSkeletonSchedule);
+        };
+    }, []);
+
+    const sectionsByTerm: [string, string[]][] = useMemo(() => {
+        const result = skeletonSchedule.courses.reduce((accumulated, course) => {
+            accumulated[course.term] ??= [];
+            accumulated[course.term].push(course.sectionCode);
+            return accumulated;
+        }, {} as Record<string, string[]>);
+
+        return Object.entries(result);
+    }, [skeletonSchedule.courses]);
+
+    return (
+        <>
```

Comment:
```suggestion
        <>
                        <Typography variant="h6">{`${skeletonSchedule.scheduleName}`}</Typography>

```

Looking at `ShortCourseScheduleSchema`, we should be able to also support displaying the name of the schedule. 

_You may need to restyle the term typography a little bit, depending on how you decide to implement the typography of the schedule name._

Reply:
In a similar vein, I think it... should be possible to display custom events and schedule notes as well?

The base functionality is great and I think adding on this additional support would really make it awesome

Reply:
Can we add that later lol; I don't wanna debug more features

Reply:
Sure, let's hold off on custom events and notes. But the one thing I will really really push for right now is having the schedule name shown.

Otherwise, there's no real way for users to figure out what's what based off just the codes.

Reply:
Did I not do that? https://github.com/icssc/AntAlmanac/blob/de7f86b5ca63b22e5bdc1f952144448ccfe2a826/apps/antalmanac/src/components/RightPane/AddedCourses/AddedCoursePane.tsx#L110

Reply:
💀 sorry, I'm on mobile and didn't see it 😅

I'll check it out once I'm home; Thanks for adding in the schedule name :))